Expensify/App
# Fix issues with network queue during sign in

### Details

Fixes uncleared listener for the App Visibility, which causes unauthorized API request and breaks the Network Queue.
For the details of the issue see [1118#issuecomment-753555165](https://github.com/Expensify/Expensify.cash/issues/1118#issuecomment-753555165).

### Fixed Issues

Fixes #1118 

### Tests

1. Login to the web app
2. Navigate to any chat
3. Click "Sign Out"
4. Switch to other tab in the browser
5. Switch back to the Sign In page
6. Wait for 3 seconds
7. Enter the email, press "Continue"
8. Verify that sign in continues and user is able to enter the password

### Tested On


### Screenshots
<Add any necessary screenshots for all platforms if your change added or updated UI>

#### Web

https://user-images.githubusercontent.com/130532/104510226-89aa8600-55f3-11eb-98fa-ebfd7296c5d9.mp4

#### Mobile Web

https://user-images.githubusercontent.com/130532/104516491-1574e000-55fd-11eb-94bc-4dc9e6f416f0.mp4

#### Desktop

https://user-images.githubusercontent.com/130532/104510280-9dee8300-55f3-11eb-9eb0-728523a1ea7f.mp4

#### iOS

https://user-images.githubusercontent.com/130532/104516719-7997a400-55fd-11eb-958d-92a877b713ed.mp4

#### Android

https://user-images.githubusercontent.com/130532/104520356-a9e24100-5603-11eb-9f3d-b0cf1223d022.mp4
